drivers/dma/dma_intel_adsp_gpdma.c

```diff
@@ -299,6 +314,21 @@ int intel_adsp_gpdma_get_attribute(const struct device *dev, uint32_t type, uint
 	return 0;
 }
 
+static int gpdma_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+	case PM_DEVICE_ACTION_RESUME:
+	case PM_DEVICE_ACTION_TURN_ON:
+	case PM_DEVICE_ACTION_TURN_OFF:
+		break;
+	default:
```

Comment:
So... this is a stub.  There's nothing in this PR that does anything real.  Surely the runtime PM layer will noop-out calls to pm_put/get already, no?  Why bother with this stuff?

Reply:
AFAIU it is empty because the only thing that you want is to hold a reference to the power domain when this device is in use to prevent the domain to suspend. So, you don't really need anything here.

If that is the case, why you are calling `pm_device_init_suspended(dev);`  if it does not do anything ? The other calls to pm_device_runtime_get/put will trigger a correspondent action in the domain and look ok.

I'm really surprised that we need to declare a callback in these situations, but that is necessary at the moment. One thing you could do is keep a reference to the domain in the driver itself and call runtime_get()/put() directly to the domain, but the way it is implemented here is ok for me.